src/java/org/apache/cassandra/service/accord/AccordLoadingState.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.accord;
+
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+
+import accord.utils.async.AsyncChain;
+import accord.utils.async.AsyncResults;
+
+/**
+ * Global state that manages loading states
+ */
+public class AccordLoadingState<K, V>
+{
+    public enum LoadingState { NOT_FOUND, PENDING, LOADED, FAILED }
```

Comment:
NOT_FOUND -> UNINITIALIZED?

I spent a while this morning thinking this meant it wasn't found on disk, and trying to figure out how this differed from LOADED.